eks/terraform/modules/cluster/main.tf

```diff
@@ -690,26 +711,34 @@ module "node_group_monitoring" {
 ################################################################################
 
 locals {
-  cluster_autoscaler_helm_values = <<HELMVALUES
-awsRegion: ${var.region}
-autoDiscovery:
-  clusterName: ${var.cluster_name}
-extraArgs:
-  scale-down-delay-after-add: 5m
-  scale-down-unneeded-time: 5m
-replicaCount: 2
-rbac:
-  serviceAccount:
-    name: ${local.cluster_autoscaler_service_account}
-    annotations:
-      eks.amazonaws.com/role-arn: ${try(module.cluster_autoscaler_irsa_role.iam_role_arn, "")}
-HELMVALUES
-
-  load_balancer_controller_helm_values = <<HELMVALUES
-clusterName: ${var.cluster_name}
-serviceAccount:
-  name: ${local.loadbalancer_controller_service_account}
-  annotations:
-    eks.amazonaws.com/role-arn: ${try(module.loadbalancer_controller_irsa_role.iam_role_arn, "")}
-HELMVALUES
+  cluster_autoscaler_helm_values = yamlencode({
+    awsRegion : var.region,
+    autoDiscovery : {
+      clusterName : var.cluster_name
+    },
+    extraArgs : {
+      "scale-down-delay-after-add" : "5m",
+      "scale-down-unneeded-time" : "5m"
+    },
+    replicaCount : 2,
+    rbac : {
+      serviceAccount : {
+        name : local.cluster_autoscaler_service_account,
+        annotations : {
+          "eks.amazonaws.com/role-arn" : try(module.cluster_autoscaler_irsa_role.iam_role_arn, "")
+        }
+      }
+    }
+  })
+
+  load_balancer_controller_helm_values = yamlencode({
+    clusterName : var.cluster_name,
+    serviceAccount : {
+      name : local.loadbalancer_controller_service_account,
+      annotations : {
+        "eks.amazonaws.com/role-arn" : try(module.loadbalancer_controller_irsa_role.iam_role_arn, "")
+      }
+    },
+    defaultTags : var.common_tags
```

Comment:
This is the only real change to this block - the rest is just switching from heredocs to yamlencode which I think is cleaner.

Reply:
Specifically the 'defaultTags' bit.